### ISSUE TITLE
libmowgli 2.1.0+ - cannot compile on Linux with musl libc

### ISSUE THREAD

**Issue by awilfox** [atheme/libmowgli-2#43]
I was attempting to package libmowgli for Adélie Linux, a new distribution which uses the musl libc for strict POSIX conformance and speed.  `_SC_LEVEL1_DCACHE_LINESIZE` is not defined by the musl libc, so the `platform` module fails to compile with this error:

```
cacheline.c: In function 'mowgli_cacheline_bootstrap':
cacheline.c:36:27: error: '_SC_LEVEL1_DCACHE_LINESIZE' undeclared (first use in this function)
  cacheline_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE);
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
cacheline.c:36:27: note: each undeclared identifier is reported only once for each function it appears in
make[3]: *** [../../../buildsys.mk:349: cacheline.lib.o] Error 1
make[2]: *** [../../buildsys.mk:114: platform] Error 2
```

Perhaps that could be checked for by autoconf and only used where available, instead of assuming all Linux systems define it?  However, [Bionic defines it as 0](https://android.googlesource.com/platform/bionic/+/master/libc/bionic/sysconf.cpp#205), so it is likely not useful there either.

**Comment by grawity** [comment]
Comments in #41 are probably relevant

**Comment by awilfox** [comment]
Wow, that's not even valid C.  However, it does give me an idea; why not change the ifdef to be
```c
#if defined(MOWGLI_OS_LINUX) && defined(_SC_LEVEL1_DCACHE_LINESIZE)
```
?

**Comment by kaniini** [comment]
just remove it imo

**Comment by alyx** [comment]
Resolved in #44
